src/react/DevicesPage.js

```diff
@@ -0,0 +1,353 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2020 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import React from "react";
+import {
+  Container,
+  Row,
+  Col,
+  Button,
+  OverlayTrigger,
+  Pagination,
+  Spinner,
+  Table,
+  Tooltip
+} from "react-bootstrap";
+
+import AstarteClient from "./AstarteClient.js";
+import Device from "./astarte/Device.js";
+import SingleCardPage from "./ui/SingleCardPage.js";
+import { Link } from "react-router-dom";
+
+const DEVICES_PER_PAGE = 20;
+const MAX_SHOWN_PAGES = 10;
+
+export default class DevicesPage extends React.Component {
+  constructor(props) {
+    super(props);
+
+    let config = JSON.parse(localStorage.session).api_config;
+    let protocol = config.secure_connection ? "https://" : "http://";
```

Comment:
TODO: let's add config.http_protocol() function that returns https or http according to config.secure_connection.